educator/src/Dscp/Educator/Web/Student/Auth.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE DataKinds     #-}
+-- | Nessesary types and implementation for Student authenthication
+
+module Dscp.Educator.Web.Student.Auth
+       ( StudentAuth
+       , WithStudent (..)
+       , AuthData (..)
+       , GetStudents
+       ) where
+
+import Crypto.JOSE.JWK hiding (hash)
+import Crypto.JWT hiding (hash)
+import Data.Aeson
+import Servant.Auth.Server
+import Servant.Auth.Server.Internal.Class
```

Comment:
Please, don't import modules outside of the project like that, _especially_ `Internal` ones.
If possible, don't import `Internal` modules at all.
If you import, either put a list of names to import or do it in qualified way.

Reply:
Yeah, I didn't leave a comment on it, but unqualified imports from external libraries are considered a bad practice, so please spend several minutes to make them qualified.